### PR TITLE
Added full fan speed toggle

### PR DESCRIPTION
**Pull request by m33ts4k0z**
## Summary by CodeRabbit

- **New Features**
  - Introduced a new fan control feature allowing users to set the fan to full speed directly from the device settings.

- **User Interface Enhancements**
  - Added a new toggle switch in the device settings page for easy control of the fan speed.

- **Localization**
  - Included new localized descriptions and text for the fan speed settings feature.

- **Bug Fixes**
  - Ensured fan speed is reset to default on device shutdown or restart for better device management.

### CODE REVIEW

File: HandheldCompanion/Views/QuickPages/QuickDevicePage.xaml.cs [hunk 1]
```diff
@@ -89,4 +92,22 @@ private void ComboBoxOverlayDisplayLevel_SelectionChanged(object sender, Selecti
...
+
+        if (isSwitchOn)
+        {
+            device.SetFanFullSpeed(1);
+        }
+        else
+        {
+            device.SetFanFullSpeed(0);
+        }
+    }
```
[the remark this answers is not in the document]
Comment (MSeys) [reply HandheldCompanion/Views/QuickPages/QuickDevicePage.xaml.cs:112]: This does seem a bit weird indeed. Shouldn't the current device be retrieved and checked whether it is a Legion Go device and then call the function?

File: HandheldCompanion/Devices/Lenovo/LegionGo.cs [hunk 2]
```diff
@@ -132,6 +132,12 @@ public override void SetFanControl(bool enable, int mode = 0)
         // do something
     }
 
+    public void SetFanFullSpeed(byte enable = 0)
+    {
+        // Fan control: Default, Full (0, 1)
+        ECRAMWrite(0x8A, enable);
+    }
```
[the remark this answers is not in the document]
Comment (MSeys) [reply HandheldCompanion/Devices/Lenovo/LegionGo.cs:139]: Not the biggest fan of this. Can the parameter be a bool instead? (Also don't see the use of a default value)
Cast bool to byte if necessary for the function itself or do if-check in there. (E.g. bool ? 1 : 0)

[the remark this answers is not in the document]
Comment (MSeys) [reply HandheldCompanion/Devices/Lenovo/LegionGo.cs:139]: (byte)(enabled ? 1 : 0) should also be possible, I believe.

File: HandheldCompanion/Views/QuickPages/QuickDevicePage.xaml.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (MSeys) [reply HandheldCompanion/Views/QuickPages/QuickDevicePage.xaml.cs:112]: I would also change this to also use the function that accepts the bool parameter instead.

File: HandheldCompanion/Views/QuickPages/QuickDevicePage.xaml [hunk 3]
```diff
@@ -67,5 +67,35 @@
                 </ComboBox>
             </Border>
         </ui:SimpleStackPanel>
+        <ui:SimpleStackPanel Spacing="6" Name="FullFanSpeedToggler">
```
Comment (MSeys) [inline HandheldCompanion/Views/QuickPages/QuickDevicePage.xaml:70]: Where's the check to hide this for non Legion Go Devices?

**Review by MSeys — CHANGES_REQUESTED** [review]

File: HandheldCompanion/Views/QuickPages/QuickDevicePage.xaml.cs [hunk 4]
```diff
@@ -21,6 +23,7 @@ public QuickDevicePage()
     {
         InitializeComponent();
 
+        FullFanSpeedToggler.Visibility = MainWindow.CurrentDevice is LegionGo ? Visibility.Visible : Visibility.Hidden;
```
Comment (m33ts4k0z) [inline HandheldCompanion/Views/QuickPages/QuickDevicePage.xaml.cs:26]: @MSeys The check to hide if for non LeGo devices is actually here. Should I move it?

**Comment by Valkirie** [comment]
Good job team ! Merging.

**Pull request merged**